Tests/_InternalTestSupportTests/Misc.swift

```diff
@@ -0,0 +1,124 @@
+
+        let actual = getNumberOfMatches(of: matchOn, in: value)
+
+        XCTAssertEqual(actual, expectedNumMatches, "Actual is not as expected")
+    }
+
+    func testNonEmptyStringMatchesOnEmptyStringReturnsZero()  {
+        let matchOn = """
+        This is a non-empty string
+        This is the second line
+        This is the third line
+        """
+        let value = ""
+        let expectedNumMatches = 0
+
+        let actual = getNumberOfMatches(of: matchOn, in: value)
+
+        XCTAssertEqual(actual, expectedNumMatches, "Actual is not as expected")
+    }
+
+    func testfatalErrorMatchesOnMultiLineWithTwoOccurencesReturnsTwo() {
+        let matchOn = "error: fatalError"
+        let value = """
+        > swift test                                                                                          25/10/24 10:44:14
+        Building for debugging...
+        /Users/arandomuser/Documents/personal/repro-swiftpm-6605/Tests/repro-swiftpm-6605Tests/repro_swiftpm_6605Tests.swift:7:19: error: division by zero
+                let y = 1 / x
+                        ^
+        error: fatalError
+
+        error: fatalError
+        """
+        let expectedNumMatches = 2
+
+        let actual = getNumberOfMatches(of: matchOn, in: value)
+
+        XCTAssertEqual(actual, expectedNumMatches, "Actual is not as expected")
+    }
+
+    func testfatalErrorWithLeadingNewLineMatchesOnMultiLineWithTwoOccurencesReturnsTwo() {
+        let matchOn = "\nerror: fatalError"
+        let value = """
+        > swift test                                                                                          25/10/24 10:44:14
+        Building for debugging...
+        /Users/arandomuser/Documents/personal/repro-swiftpm-6605/Tests/repro-swiftpm-6605Tests/repro_swiftpm_6605Tests.swift:7:19: error: division by zero
+                let y = 1 / x
+                        ^
+        error: fatalError
+
+        error: fatalError
+        """
+        let expectedNumMatches = 2
+
+        let actual = getNumberOfMatches(of: matchOn, in: value)
+
+        XCTAssertEqual(actual, expectedNumMatches, "Actual is not as expected")
+    }
+
+    func testfatalErrorWithLeadingAndTrailingNewLineMatchesOnMultiLineWithOneOccurencesReturnsOne() {
+        let matchOn = "\nerror: fatalError\n"
+        let value = """
+        > swift test                                                                                          25/10/24 10:44:14
+        Building for debugging...
+        /Users/arandomuser/Documents/personal/repro-swiftpm-6605/Tests/repro-swiftpm-6605Tests/repro_swiftpm_6605Tests.swift:7:19: error: division by zero
+                let y = 1 / x
+                        ^
+        error: fatalError
+
+        error: fatalError
+        """
+        let expectedNumMatches = 1
+
+        let actual = getNumberOfMatches(of: matchOn, in: value)
+
+        XCTAssertEqual(actual, expectedNumMatches, "Actual is not as expected")
+    }
+
+
+
```

Comment:
Formatting is off here, would you mind running the formatter on newly added files?

Reply:
I still see stray whitespaces surrounding curly braces. Unfortunately, `swift-format` is not great at following our formatting style and lacks required customizations to specify it. We're using https://github.com/nicklockwood/SwiftFormat in SwiftPM instead.

Reply:
Also see "Creating Pull Requests" section here https://github.com/swiftlang/swift-package-manager/blob/main/CONTRIBUTING.md#creating-pull-requests

Reply:
formatting all touched file using https://github.com/nicklockwood/SwiftFormat resulted in many changes. so I only formatted this file.